### PR TITLE
fix some grok filter issues

### PR DESCRIPTION
**Pull request by bloodearnest**
Allow _ in logger name, and handle any foo= values by setting them to empty string, as the kv filter can't parse them properly

### CODE REVIEW

File: docs/logging.rst [hunk 1]
```diff
@@ -209,24 +209,26 @@ line, following the `logfmt <https://brandur.org/logfmt>`_ idea. e.g.::
...
 
-    Both keys and values can be of arbitrary length, and either utf8 encoded
-    bytes, or unicode. Talisker will always encode the output in utf8.
+    Both keys and values are encoded in utf8, and truncated after 2kb, to avoid
+    accidental hugh log messages.
 
-    The reason for stripping " characters is to do with the limitations of
-    logstash's kv filter, which cannot currently cope with them, even when
-    escaped. See `issue 2
+    The
```
Comment (wgrant) [inline docs/logging.rst]: Not meant to be here?

File: talisker/logs.py [hunk 2]
```diff
@@ -318,32 +325,41 @@ def remove_quotes(self, s):
...
-        if isinstance(k, bytes):
-            k = k.decode('utf8')
+        # we need unicode strings so as to be able to do replace
+        if not v:
+            return None
+
         if not isinstance(v, str):
             if isinstance(v, bytes):
                 v = v.decode('utf8')
             elif not isinstance(v, str):
```
Comment (wgrant) [inline talisker/logs.py]: Isn't this just "else", given it's already inside an if block with the same predicate?

File: docs/logging.rst (hunk 1 above)
Comment (bloodearnest) [inline docs/logging.rst]: yep, gone

File: talisker/logs.py (hunk 2 above)
Comment (bloodearnest) [inline talisker/logs.py]: Indeed it is, gone

**Pull request merged**